### PR TITLE
Add 'bluesky-queueserver'

### PR DESCRIPTION
**Pull request by dmgav**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/bluesky-queueserver/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    - qserver-zmq-keys --help
+    - qserver-console-monitor --help
+  requires:
+    - pip
+
+about:
+  home: https://github.com/bluesky/bluesky-queueserver
+  summary: Server for queueing plans
+  license: BSD-3-Clause
+  license_file: LICENSE
```
Comment (mrakitin) [inline recipes/bluesky-queueserver/meta.yaml:69]:
```suggestion
  license_file: LICENSE
  doc_url: https://blueskyproject.io/bluesky-queueserver
  dev_url: https://github.com/bluesky/bluesky-queueserver
```

File: recipes/bluesky-queueserver/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/bluesky-queueserver-{{ version }}.tar.gz
+  sha256: 07e0a0a065e96c9bc378bda60dd99987d0481951e86ec601152b14ae3a915847
+
+build:
+  number: 0
+  skip: true   # [py<37]
```
Comment (mrakitin) [inline recipes/bluesky-queueserver/meta.yaml:15]:
```suggestion
  skip: true   # [py<37 or win]
```

**Review by mrakitin — APPROVED** [review]
Looks good to me. A couple of minor updates + exclude Windows builds as the platform is not supported by the package.

@conda-forge/staged-recipes, I agree to be listed as a maintainer.

File: recipes/bluesky-queueserver/meta.yaml (hunk 2 above)
Comment (mrakitin) [reply recipes/bluesky-queueserver/meta.yaml:15]: As discussed with @dmgav over DM, this case may be needed, so I'll retract this suggestion.

File: recipes/bluesky-queueserver/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  run:
+    - aioredis <2.0.0
+    - bluesky
+    - bluesky-kafka
+    - databroker
+    - event-model
+    - json-rpc
+    - jsonschema
+    - matplotlib-base
+    - msgpack >=1.0.0
```
Comment (dmgav) [inline recipes/bluesky-queueserver/meta.yaml:37]:
```suggestion
    - msgpack-python >=1.0.0
```

File: recipes/bluesky-queueserver/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - requests
+    - scikit-image
+    - typing-extensions  # [py<38]
+
+test:
+  imports:
+    - bluesky_queueserver
+    - bluesky_queueserver.manager
+  commands:
+    - pip check
```
Comment (dmgav) [inline recipes/bluesky-queueserver/meta.yaml:56]:
```suggestion
```

File: recipes/bluesky-queueserver/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/bluesky-queueserver-{{ version }}.tar.gz
+  sha256: 07e0a0a065e96c9bc378bda60dd99987d0481951e86ec601152b14ae3a915847
+
+build:
+  number: 0
+  skip: true   # [py<37]
```
Comment (dmgav) [inline recipes/bluesky-queueserver/meta.yaml:15]:
```suggestion
  skip: true   # [py<37 or win]
```

File: recipes/bluesky-queueserver/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - aioredis <2.0.0
+    - bluesky
+    - bluesky-kafka
```
Comment (mrakitin) [inline recipes/bluesky-queueserver/meta.yaml:31]:
```suggestion
    - bluesky-kafka  # [not win]
```

File: recipes/bluesky-queueserver/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/bluesky-queueserver-{{ version }}.tar.gz
+  sha256: 07e0a0a065e96c9bc378bda60dd99987d0481951e86ec601152b14ae3a915847
+
+build:
+  number: 0
+  skip: true   # [py<37 or win]
```
Comment (mrakitin) [inline recipes/bluesky-queueserver/meta.yaml:15]:
```suggestion
  skip: true   # [py<37]
```

**Review by mrakitin — COMMENTED** [review]
Here is a possible workaround for the builds on Windows.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, @dopplershift, this is ready for your review! Could you please review/merge soon?

File: recipes/bluesky-queueserver/meta.yaml (hunk 2 above)
Comment (goanpeca) [reply recipes/bluesky-queueserver/meta.yaml:15]: Why is this needed @mrakitin ?

Also, was it supposed to skip `[win]`?

Thanks!

Comment (mrakitin) [reply recipes/bluesky-queueserver/meta.yaml:15]: The package support is for Python 3.7+ only, so with this selector we exclude currently available Python 3.6. We do not need to exclude the Windows platform as a part of the package's functionality works in Windows. We excluded the `bluesky-kafka` from the `run` dependencies list that depends on `librdkafka` that exists only for unix platforms. The tip of this PR branch currently has all needed fixes.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, is there anything that has to be done before this PR gets merged? Thanks for your help!

**Review by dopplershift — APPROVED** [review]

**Comment by mrakitin** [comment]
Thanks a lot, @dopplershift!

**Pull request merged**